lib/puma/server.rb

```diff
@@ -896,11 +900,16 @@ def notify_safely(message)
       begin
         @notify << message
       rescue IOError
+        Thread.current.purge_interrupt_queue if Thread.current.respond_to? :purge_interrupt_queue
         # The server, in another thread, is shutting down
       rescue RuntimeError => e
         # The server, in another thread, has been shut down during the system call
         # https://github.com/puma/puma/pull/1206
-        raise e unless e.message.include?('IOError')
+        if e.message.include?('IOError')
```

Comment:
This reminds me old story: admin setup script to monitor server log, if there was fatal error in logs it restarted server. Later one user write to support complaining that every time one tries to login server was down. Username was "femme fatale"